overlord/snapstate/snapstate.go

```diff
@@ -1773,6 +1811,13 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 
 		ts, err := doInstall(st, snapst, snapsup, 0, fromChange, inUseFor(deviceCtx))
 		if err != nil {
+			if errors.Is(err, &timedBusySnapError{}) && len(ts.Tasks()) > 0 {
```

Comment:
I wonder if some of the internal helpers here should produce two groups/sets of tasksets, one for predownload and one for the rest? it might make some of the code clearer in terms of the two groups going into different changes?